### PR TITLE
Disable Bitmap or Hll type in keys or in values with incorrect agg-type

### LINKED ISSUE #3783: BE will crash when execute stream load, if bitmap field does not specify BITMAP_UNION type.
BE will crash when execute stream load, if bitmap field does not specify BITMAP_UNION type.

1. bitmap field does not specify BITMAP_UNION when create table.
'''
CREATE TABLE bitmaptable
(
    TSID     BIGINT NOT NULL,
    PKEY     VARCHAR(96) NOT NULL,
    EDITFLAG BITMAP NULL
)
UNIQUE KEY(TSID,PKEY)
DISTRIBUTED BY HASH(TSID) BUCKETS 10;
'''

2. data file "111.txt" as the following:
'''
555;666;1,2,3
777;888;1,2,3
'''

3. exec stream load: 
'''
curl --location-trusted -u user:pwd -H "column_separator:;" -H "columns:TSID,PKEY,tmp,EDITFLAG=bitmap_from_string(tmp)" -T /mydata/bitmap/111.txt -H "label:job1" http://172.17.0.2:8040/api/dolap/bitmaptable/_stream_load
'''

4. BE will crash.

### PR DESCRIPTION
**Pull request by spaces-X**
Bitmap and Hll type can not be used with incorrect aggregate functions, which will cause to BE crush.
Add some logical checks in FE's ColumnDef#analyze to avoid creating tables or changing schemas incorrectly.

- Keys never be bitmap or hll type
- values with bitmap or hll type have to be associated with bitmap_union or hll_union

### CODE REVIEW

**Comment by spaces-X** [comment]
[issue](https://github.com/apache/incubator-doris/issues/3769)

**Comment by morningman** [comment]
> [issue](https://github.com/apache/incubator-doris/issues/3769)

Link issue like this: `#3769`
If you want to close the issue automatically after PR being merged, like this: `Fix #3769`

**Review by morningman — CHANGES_REQUESTED** [review]
Please add UT for this case.
You can reference to `CreateTableStmtTest.java`

**Comment by spaces-X** [comment]
[Fix #3769](https://github.com/apache/incubator-doris/issues/3769)

**Comment by spaces-X** [comment]
> Please add UT for this case.
> You can reference to `CreateTableStmtTest.java`

Done.

File: fe/src/main/java/org/apache/doris/catalog/AggregateType.java [hunk 1]
```diff
@@ -85,20 +85,22 @@
...
-        primitiveTypeList.clear();
-        compatibilityMap.put(REPLACE_IF_NOT_NULL, EnumSet.allOf(PrimitiveType.class));
-       
-        primitiveTypeList.clear();
         primitiveTypeList.add(PrimitiveType.HLL);
         compatibilityMap.put(HLL_UNION, EnumSet.copyOf(primitiveTypeList));
 
-        primitiveTypeList.clear();
+
         primitiveTypeList.add(PrimitiveType.BITMAP);
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/catalog/AggregateType.java:100]: Why remove the `primitiveTypeList.clear();`.
Now, the `BITMAP_UNION` is compatible with both `BITMAP` and `HLL`?

**Review by morningman — CHANGES_REQUESTED** [review]

File: fe/src/main/java/org/apache/doris/catalog/AggregateType.java (hunk 1 above)
Comment (spaces-X) [reply fe/src/main/java/org/apache/doris/catalog/AggregateType.java:101]: By mistake.
Thanks for reminding.

File: fe/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 2]
```diff
@@ -131,6 +131,16 @@ public void analyze(boolean isOlap) throws AnalysisException {
 
         Type type = typeDef.getType();
 
+        // disable Bitmap Hll type in keys, values without aggregate function.
+        if ( type.isBitmapType() || type.isHllType() ){
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/analysis/ColumnDef.java:135]: Should keep code style with original code.
```suggestion
        if (type.isBitmapType() || type.isHllType()) {
```

File: fe/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 3]
```diff
@@ -131,6 +131,16 @@ public void analyze(boolean isOlap) throws AnalysisException {
...
+            if (isKey){
```
Comment (imay) [inline fe/src/main/java/org/apache/doris/analysis/ColumnDef.java:136]:
```suggestion
            if (isKey) {
```

**Review by kangkaisen — APPROVED** [review]
+1, LGTM

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**